### PR TITLE
Fix Code Standards for RC 2 release

### PR DESCRIPTION
**Pull request by getdave**
## What?

Updates code for WP 6.5 RC2 destined for WP Core to align with standards.

See https://github.com/WordPress/wordpress-develop/pull/6253/

## Why?

CS

## How?

Resolve all comments from https://github.com/WordPress/wordpress-develop/pull/6253/

## Screenshots or screencast 

Co-authored-by: getdave <get_dave@git.wordpress.org>
Co-authored-by: swissspidy <swissspidy@git.wordpress.org>

### CODE REVIEW

File: packages/block-library/src/navigation/index.php
Comment (swissspidy) [inline packages/block-library/src/navigation/index.php]:
```suggestion
 * @access private
 * @since 6.5.0
 *
 * @param stdClass $post Post object.
 * @return stdClass The updated post object.
```

In line with PHP docs standards.

File: packages/block-library/src/navigation/index.php
Comment (swissspidy) [inline packages/block-library/src/navigation/index.php]: Accidentally commented out

File: packages/block-library/src/navigation/index.php
Comment (swissspidy) [inline packages/block-library/src/navigation/index.php:1556]: So all these comments are only really relevant for GB, not core, right? 'cause obviously when merging this change into core, these functions exist.

Comment (getdave) [reply packages/block-library/src/navigation/index.php:1556]: The file is maintained in Gutenberg and those comments are relevant. Whilst I appreciate they aren't relevant for Core, we don't have a process for handling exclude comments at this time.

How critical is it to resolve this for RC 2?

Can we discuss a solution for RC 3?

File: packages/block-library/src/navigation/index.php
Comment (getdave) [reply packages/block-library/src/navigation/index.php]: Fixed

File: packages/block-library/src/navigation/index.php
Comment (getdave) [inline packages/block-library/src/navigation/index.php]: @swissspidy My local PHPCS gives me a warning here. Will this also cause an Issue with Core merge?

Comment (getdave) [reply packages/block-library/src/navigation/index.php]: I disabled PHPCS for this line.

Comment (swissspidy) [reply packages/block-library/src/navigation/index.php]: @getdave what's the warning? Note that it's usually recommended to use `// phpcs:ignore` comments just for that particular warning, e.g. `// phpcs:ignore WordPress.DateTime.CurrentTimeTimestamp.Requested` or whatever. Makes it much clearer what the intention is.

File: packages/block-library/src/navigation/index.php
Comment (swissspidy) [reply packages/block-library/src/navigation/index.php:1556]: Just a general remark, not a blocker. I suppose we could remove remove these hardening checks & comments in the future though, e.g. in 6.6?

File: packages/block-library/src/navigation/index.php
Comment (getdave) [reply packages/block-library/src/navigation/index.php]: > String concat is not required here; use a single string instead

All I get is that

Comment (getdave) [reply packages/block-library/src/navigation/index.php]: I think the rule might be `Generic.Strings.UnnecessaryStringConcat.Found`.

Comment (getdave) [reply packages/block-library/src/navigation/index.php]: ✅

File: packages/block-library/src/navigation/index.php
Comment (youknowriad) [reply packages/block-library/src/navigation/index.php:1556]: I think at some point we should gitignore these files in Core, they should be considered a "library" like any JS library. But I guess that change can only happen when/if we embrace php packages as well.

**Comment by getdave** [comment]
Just waiting on an ✅ Approval 🙏

**Review by ockham — APPROVED** [review]
Thank you!

**Comment by getdave** [comment]
@youknowriad I think we need to get this merged into order to make the release code freeze. Any further followups will need to wait until RC3 or we can discuss delaying the release.

**Comment by youknowriad** [comment]
Merging, we don't have much time left before the release.

**Comment by getdave** [comment]
I'll cherry pick this to the wp branch and re-run the release

File: packages/block-library/src/navigation/index.php
Comment (getdave) [reply packages/block-library/src/navigation/index.php:1556]: I raised [this Discussion](https://github.com/WordPress/gutenberg/discussions/59786) in the hope we can improve the whole process for further releases.

Comment (ockham) [reply packages/block-library/src/navigation/index.php:1556]: As for the Block Hooks related logic found in the Navigation block code specifically, we might be able to absorb it into the more generic `inject_ignored_hooked_blocks_metadata_attributes` function present in Core, see https://core.trac.wordpress.org/ticket/60759.

**Pull request merged**
